app/controllers/AddAnotherDocumentController.scala

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers
+
+import config.FrontendAppConfig
+import controllers.actions._
+import forms.AddAnotherFormProvider
+import models.{LocalReferenceNumber, Mode, NormalMode}
+import navigation.DocumentsNavigatorProvider
+import play.api.data.Form
+import play.api.i18n.{I18nSupport, MessagesApi}
+import play.api.mvc.{Action, AnyContent, MessagesControllerComponents}
+import repositories.SessionRepository
+import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendBaseController
+import viewModels.AddAnotherDocumentViewModel
+import viewModels.AddAnotherDocumentViewModel.AddAnotherDocumentViewModelProvider
+import views.html.AddAnotherDocumentView
+
+import javax.inject.Inject
+import scala.concurrent.Future
+
+class AddAnotherDocumentController @Inject() (
+  override val messagesApi: MessagesApi,
+  implicit val sessionRepository: SessionRepository,
+  navigatorProvider: DocumentsNavigatorProvider,
+  actions: Actions,
+  formProvider: AddAnotherFormProvider,
+  viewModelProvider: AddAnotherDocumentViewModelProvider,
+  val controllerComponents: MessagesControllerComponents,
+  view: AddAnotherDocumentView
+)(implicit config: FrontendAppConfig)
+    extends FrontendBaseController
+    with I18nSupport {
+
+  private def form(viewModel: AddAnotherDocumentViewModel): Form[Boolean] =
+    formProvider(viewModel.prefix, viewModel.allowMore)
+
+  def onPageLoad(lrn: LocalReferenceNumber, mode: Mode): Action[AnyContent] = actions.requireData(lrn).async {
+    implicit request =>
+      val viewModel = viewModelProvider(request.userAnswers, mode)
+
+      viewModel.count match {
+        case 0 => Future.successful(Redirect(navigatorProvider(mode).nextPage(request.userAnswers)))
+        case _ => Future.successful(Ok(view(form(viewModel), lrn, viewModel)))
+      }
+  }
+
+  def onSubmit(lrn: LocalReferenceNumber, mode: Mode): Action[AnyContent] = actions.requireData(lrn) {
+    implicit request =>
+      val viewModel = viewModelProvider(request.userAnswers, mode)
+      form(viewModel)
+        .bindFromRequest()
+        .fold(
+          formWithErrors => BadRequest(view(formWithErrors, lrn, viewModel)),
+          {
+            case true  => Redirect(controllers.document.routes.TypeController.onPageLoad(lrn, NormalMode, viewModel.nextIndex))
```

Comment:
Is this correct?

Reply:
I believe so, the onPageLoad would redirect to the relevant starting page if they have 0 documents added. If they land on the page, they must at least have 1 document added and after the first index it's always the Type page

Reply:
Ah yes, I see that on the drawio now. Good shout. Marking this as resolved.